crates/api_models/src/payments.rs

```diff
@@ -1145,8 +1146,23 @@ impl GetAddressFromPaymentMethodData for Card {
 }
 
 impl Card {
-    fn apply_additional_card_info(&self, additional_card_info: AdditionalCardInfo) -> Self {
-        Self {
+    fn apply_additional_card_info(
+        &self,
+        additional_card_info: AdditionalCardInfo,
+    ) -> Result<Self, error_stack::Report<ValidationError>> {
+        let card_network = self
+            .card_network
+            .clone()
+            .or_else(|| additional_card_info.card_network.clone())
+            .map(|network| match self.card_number.is_cobadged_card() {
```

Comment:
can use `and_then` to avoid flatten

Reply:
We aren't using `and_then` here because the closure's returns `Option<Result<Option<..>, error>>`